src/main/java/org/jenkinsci/plugins/docker/commons/fingerprint/DockerFingerprintAction.java

```diff
@@ -0,0 +1,141 @@
+import org.jenkinsci.plugins.docker.commons.Messages;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+/**
+ * Adds a link, which provides a list of fingerprints with
+ * {@link DockerRunPtrFingerprintFacet}s.
+ *
+ * @author Oleg Nenashev
+ */
+public class DockerFingerprintAction implements RunAction2 {
+
+    private Set<String> imageIDs;
+    transient Run<?, ?> run;
+
+    public DockerFingerprintAction() {
+        this.imageIDs = new HashSet<String>();
+    }
+    
+    public Object readResolve() {
+        if (imageIDs == null) {
+            imageIDs = new HashSet<String>();
+        }
+        return this;
+    } 
+
+    public Run<?, ?> getRun() {
+        return run;
+    }
+
+    @Override
+    public void onAttached(Run<?, ?> r) {
+        run = r;
+    }
+
+    @Override
+    public void onLoad(Run<?, ?> r) {
+        run = r;
+    }
+
+    @Override
+    public String getIconFileName() {
+        return "fingerprint.png";
+    }
+
+    @Override
+    public String getDisplayName() {
+        return Messages.DockerFingerprintAction_DisplayName();
+    }
+
+    @Override
+    public String getUrlName() {
+        return "docker-fingerprints";
+    }
+
+    /**
+     * Gets a list of fingerprints related to the action
+     *
+     * @return Unmodifiable set of items.
+     */
+    public Set<String> getImageIDs() {
+        return Collections.unmodifiableSet(imageIDs);
+    }
+
+    @Restricted(NoExternalUse.class)
+    public @CheckForNull String getFingerprintHash(@CheckForNull String imageId) {
+        return (imageId != null) ? DockerFingerprints.getImageFingerprintHash(imageId) : null;
+    }
+    
+    @Restricted(NoExternalUse.class)
+    public @CheckForNull Fingerprint getFingerprint(@CheckForNull String imageId) {
+        if (imageId == null) {
+            return null;
+        }
+        
+        try {
+            return DockerFingerprints.of(imageId);
+        } catch (IOException ex) {
+            return null; // nothing to do in web UI - return null as well
+        }
+    }
+
+    /**
+     * Adds an action with a reference to fingerprint if required. It's
+     * recommended to call the method from {
+     *
+     * @BulkChange} transaction to avoid saving the {@link Run} multiple times.
+     * @param fp Fingerprint
+     * @param imageId ID of the docker image
+     * @param run Run to be updated
+     * @throws IOException Cannot save the action
+     */
+    static void addToRun(Fingerprint fp, String imageId, Run run) throws IOException {
+        synchronized (run) {
+            DockerFingerprintAction action = run.getAction(DockerFingerprintAction.class);
+            if (action == null) {
+                action = new DockerFingerprintAction();
+                run.addAction(action);
+            }
+            
+            if (action.imageIDs.add(imageId)) {
+                run.save();
+            } // else no need to save updates
```

Comment:
In the current plugin implementation <code>BulkChange</code> is being created on the upper level